### PR TITLE
Support for FoundationDB 7.1

### PR DESCRIPTION
**Pull request by PierreZ**

### CODE REVIEW

File: foundationdb/src/future.rs [hunk 1]
```diff
@@ -685,3 +698,261 @@ impl fmt::Debug for FdbKey {
...
+        pub fn begin_range(&self) -> &[u8] {
+            unsafe {
+                std::slice::from_raw_parts(
+                    self.0.getRange.begin.key.key as *const u8,
+                    self.0.getRange.begin.key.key_length as usize,
+                )
+            }
+        }
+
+        pub fn end_range(&self) -> &[u8] {
```
Comment (Speedy37) [inline foundationdb/src/future.rs:803]: Should this returns a KeySelector ?

File: foundationdb/src/future.rs [hunk 2]
```diff
@@ -685,3 +698,261 @@ impl fmt::Debug for FdbKey {
...
+        pub fn parent_value(&self) -> &[u8] {
+            unsafe {
+                std::slice::from_raw_parts(
+                    self.0.value.key as *const u8,
+                    self.0.value.key_length as usize,
+                )
+            }
+        }
+
+        pub fn begin_range(&self) -> &[u8] {
```
Comment (Speedy37) [inline foundationdb/src/future.rs:794]: Should this returns a KeySelector ?

File: foundationdb/src/future.rs [hunk 3]
```diff
@@ -685,3 +698,261 @@ impl fmt::Debug for FdbKey {
...
+        }
+    }
+    impl PartialEq for FdbMappedValue {
+        fn eq(&self, other: &Self) -> bool {
+            self.deref() == other.deref()
+        }
+    }
+    impl Eq for FdbMappedValue {}
+
+    pub struct FdbMappedValue {
```
Comment (Speedy37) [inline foundationdb/src/future.rs:885]: A bit weird to declare it after its impls

**Review by Speedy37 — APPROVED** [review]

**Comment by Speedy37** [comment]
I'm still not features/macros are the best way to handle fdb versions.

I wonder if version features should be additives, 

For example:
 710, adds 700, 630, 620, 610, 600, 520, 510
 
In the code we would have to care about cfg! ordering and exclusion.

Example:


```rust
// From v600 to v630 included (i.e v600 to v700 not included)
cfg!(all(feature = "fdb-6_0", not(feature = "fdb-7_0")))
// v600 only
cfg!(all(feature = "fdb-6_0", not(feature = "fdb-6_1")))
```

**Comment by PierreZ** [comment]
PR ready to be reviewed,I will add tenant support in a separate PR

**Pull request merged**